library/Fission/User/Retriever/Class.hs

```diff
@@ -0,0 +1,22 @@
+module Fission.User.Retriever.Class (Retriever (..)) where
+
+import           Database.Esqueleto hiding ((=.), update)
+import qualified Database.Persist as P
+
+import           Fission.Models
+import           Fission.Prelude
+
+class Monad m => Retriever m where
+  getByUsername      :: Text          -> m (Maybe (Entity User))
+  getByHerkouAddOnId :: HerokuAddOnId -> m (Maybe (Entity User))
+
+instance MonadIO m => Retriever (Transaction m) where
+  getByUsername username = selectFirst
+    [ UserUsername P.==. username
+    , UserActive   P.==. True
+    ] []
+
+  getByHerkouAddOnId addOnId = selectFirst
+    [ UserHerokuAddOnId P.==. Just addOnId
+    , UserActive        P.==. True
```

Comment:
Because both Esqueleto and Persistent have `==.` operators, but they work on different types. To use this style, we need the Persistent `==.`, but want Esqueleto elsewhere.